src/main/java/ch/njol/skript/expressions/ExprNamed.java

```
@@ -35,51 +35,85 @@
-		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.PROPERTY,
+		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.COMBINED,
 				"%itemtype/inventorytype% (named|with name[s]) %string%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<String> name;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[0]);
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't create a '" + literal.toString() + "' inventory. It's not creatable!");
+				return false;
+			}
+		}
 		return true;
 	}
-	
+
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		String name = this.name.getSingle(e);
+	protected Object[] get(Event event, Object[] source) {
+		String name = this.name.getSingle(event);
 		if (name == null)
-			return get(source, obj -> obj); // No name provided, do nothing
+			return get(source, object -> {
+				if (object instanceof InventoryType) {
+					try {
+						return Bukkit.createInventory(null, (InventoryType) object);
+					} catch (NullPointerException | IllegalArgumentException e) {
+						// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+						// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+						// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+						Skript.error("You can't create a '" + Classes.toString((InventoryType) object) + "' inventory. It's not creatable!");
```

Why is this erroring at runtime? It should just silently fail as all the other expressions